src/Netscript/ScriptDeath.ts

```diff
@@ -3,28 +3,44 @@ import { WorkerScript } from "./WorkerScript";
 /**
  * Script death marker.
  *
+ * This is thrown in places where a script has been killed, but some running
+ * code is attempting to invoke one of its ns functions. By extending Error,
```

Comment:
If you're going to document this, let's be more comprehensive.
```suggestion
 * This is thrown in various places when a script has been killed, as opposed to ending
 * normally. It can be thrown from exit() and spawn(), which inherently kill the script,
 * or from an await if the script is killed while waiting for a ns function, or directly
 * if the script is dead but running code is still trying to execute ns functions. By extending Error,
```